2025-02-06 22:04:24.646 +02:00 [INF] 🚀 Starting POS API...
2025-02-06 22:04:24.857 +02:00 [INF] User profile is available. Using 'C:\Users\pinto\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
2025-02-06 22:04:25.011 +02:00 [ERR] Hosting failed to start
System.IO.IOException: Failed to bind to address http://127.0.0.1:5107: address already in use.
 ---> Microsoft.AspNetCore.Connections.AddressInUseException: Only one usage of each socket address (protocol/network address/port) is normally permitted.
 ---> System.Net.Sockets.SocketException (10048): Only one usage of each socket address (protocol/network address/port) is normally permitted.
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, Boolean disconnectOnFailure, String callerName)
   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Bind(EndPoint localEP)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportOptions.CreateDefaultBoundListenSocket(EndPoint endpoint)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.Bind()
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.Bind()
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportFactory.BindAsync(EndPoint endpoint, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.TransportManager.BindAsync(EndPoint endPoint, ConnectionDelegate connectionDelegate, EndpointConfig endpointConfig, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.<>c__DisplayClass28_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.LocalhostListenOptions.BindAsync(AddressBindContext context, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(ListenOptions[] listenOptions, AddressBindContext context, Func`2 useHttps, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
   at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
2025-02-06 22:05:04.375 +02:00 [INF] 🚀 Starting POS API...
2025-02-06 22:05:04.431 +02:00 [INF] User profile is available. Using 'C:\Users\pinto\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
2025-02-06 22:05:04.490 +02:00 [INF] Now listening on: http://localhost:5107
2025-02-06 22:05:04.493 +02:00 [INF] Application started. Press Ctrl+C to shut down.
2025-02-06 22:05:04.494 +02:00 [INF] Hosting environment: Development
2025-02-06 22:05:04.494 +02:00 [INF] Content root path: C:\Users\pinto\Documents\Development\POS\pos-backend
2025-02-06 22:09:51.344 +02:00 [INF] Application is shutting down...
2025-02-06 22:13:19.408 +02:00 [INF] 🚀 Starting POS API...
2025-02-06 22:13:19.483 +02:00 [INF] User profile is available. Using 'C:\Users\pinto\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
2025-02-06 22:13:19.630 +02:00 [INF] Now listening on: https://localhost:7005
2025-02-06 22:13:19.631 +02:00 [INF] Now listening on: http://localhost:5107
2025-02-06 22:13:19.699 +02:00 [INF] Application started. Press Ctrl+C to shut down.
2025-02-06 22:13:19.701 +02:00 [INF] Hosting environment: Development
2025-02-06 22:13:19.702 +02:00 [INF] Content root path: C:\Users\pinto\Documents\Development\POS\pos-backend
2025-02-06 22:13:20.448 +02:00 [INF] Request starting HTTP/2 GET https://localhost:7005/swagger - null null
2025-02-06 22:13:20.448 +02:00 [INF] Request starting HTTP/2 GET https://localhost:7005/swagger/index.html - null null
2025-02-06 22:13:20.548 +02:00 [INF] Request finished HTTP/2 GET https://localhost:7005/swagger - 499 null null 102.4109ms
2025-02-06 22:13:20.586 +02:00 [INF] Request finished HTTP/2 GET https://localhost:7005/swagger/index.html - 200 null text/html;charset=utf-8 140.4936ms
2025-02-06 22:13:21.128 +02:00 [INF] Request starting HTTP/2 GET https://localhost:7005/_vs/browserLink - null null
2025-02-06 22:13:21.128 +02:00 [INF] Request starting HTTP/2 GET https://localhost:7005/_framework/aspnetcore-browser-refresh.js - null null
2025-02-06 22:13:21.128 +02:00 [INF] Request starting HTTP/2 GET https://localhost:7005/swagger/swagger-ui.css - null null
2025-02-06 22:13:21.136 +02:00 [INF] Request finished HTTP/2 GET https://localhost:7005/_framework/aspnetcore-browser-refresh.js - 200 13756 application/javascript; charset=utf-8 8.7944ms
2025-02-06 22:13:21.146 +02:00 [INF] Sending file. Request path: '/swagger-ui.css'. Physical path: 'N/A'
2025-02-06 22:13:21.148 +02:00 [INF] Request finished HTTP/2 GET https://localhost:7005/swagger/swagger-ui.css - 499 143943 text/css 19.8125ms
2025-02-06 22:13:21.169 +02:00 [INF] Request finished HTTP/2 GET https://localhost:7005/_vs/browserLink - 200 null text/javascript; charset=UTF-8 41.3382ms
2025-02-06 22:13:21.214 +02:00 [INF] Request starting HTTP/2 GET https://localhost:7005/swagger/v1/swagger.json - null null
2025-02-06 22:13:21.224 +02:00 [INF] Request finished HTTP/2 GET https://localhost:7005/swagger/v1/swagger.json - 200 null application/json;charset=utf-8 10.2364ms
2025-02-06 22:37:58.780 +02:00 [INF] Request starting HTTP/2 POST https://localhost:7005/api/auth/refresh - null 0
2025-02-06 22:37:58.876 +02:00 [INF] CORS policy execution failed.
2025-02-06 22:37:58.877 +02:00 [INF] Request origin https://localhost:7005 does not have permission to access the resource.
2025-02-06 22:37:59.022 +02:00 [INF] Executing endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:37:59.080 +02:00 [INF] Route matched with {action = "RefreshToken", controller = "Auth"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] RefreshToken() on controller AuthController (PosBackend).
2025-02-06 22:37:59.090 +02:00 [INF] Executed action AuthController.RefreshToken (PosBackend) in 5.4907ms
2025-02-06 22:37:59.092 +02:00 [INF] Executed endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:37:59.117 +02:00 [INF] Request finished HTTP/2 POST https://localhost:7005/api/auth/refresh - 500 null application/json 341.564ms
2025-02-06 22:38:29.524 +02:00 [INF] Request starting HTTP/2 POST https://localhost:7005/api/auth/refresh - null 0
2025-02-06 22:38:29.531 +02:00 [INF] CORS policy execution failed.
2025-02-06 22:38:29.532 +02:00 [INF] Request origin https://localhost:7005 does not have permission to access the resource.
2025-02-06 22:38:29.535 +02:00 [INF] Executing endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:38:29.536 +02:00 [INF] Route matched with {action = "RefreshToken", controller = "Auth"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] RefreshToken() on controller AuthController (PosBackend).
2025-02-06 22:38:29.538 +02:00 [INF] Executed action AuthController.RefreshToken (PosBackend) in 0.2186ms
2025-02-06 22:38:29.539 +02:00 [INF] Executed endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:38:29.541 +02:00 [INF] Request finished HTTP/2 POST https://localhost:7005/api/auth/refresh - 500 null application/json 16.6593ms
2025-02-06 22:45:43.748 +02:00 [INF] Request starting HTTP/2 POST https://localhost:7005/api/auth/refresh - null 0
2025-02-06 22:45:43.790 +02:00 [INF] CORS policy execution failed.
2025-02-06 22:45:43.791 +02:00 [INF] Request origin https://localhost:7005 does not have permission to access the resource.
2025-02-06 22:45:44.441 +02:00 [INF] Failed to validate the token.
Microsoft.IdentityModel.Tokens.SecurityTokenInvalidIssuerException: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
   at Microsoft.IdentityModel.Tokens.Validators.ValidateIssuerAsync(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.Tokens.Validators.ValidateIssuer(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.Tokens.InternalValidators.ValidateAfterSignatureFailed(SecurityToken securityToken, Nullable`1 notBefore, Nullable`1 expires, IEnumerable`1 audiences, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateSignature(JsonWebToken jwtToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateSignatureAndIssuerSecurityKey(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateJWSAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
2025-02-06 22:45:44.452 +02:00 [ERR] ❌ Authentication failed: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
2025-02-06 22:45:44.457 +02:00 [INF] Bearer was not authenticated. Failure message: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
2025-02-06 22:45:44.461 +02:00 [INF] Executing endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:45:44.468 +02:00 [INF] Route matched with {action = "RefreshToken", controller = "Auth"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] RefreshToken() on controller AuthController (PosBackend).
2025-02-06 22:45:44.473 +02:00 [INF] Executed action AuthController.RefreshToken (PosBackend) in 2.6732ms
2025-02-06 22:45:44.476 +02:00 [INF] Executed endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:45:44.485 +02:00 [ERR] An unhandled exception has occurred while executing the request.
System.InvalidOperationException: StatusCode cannot be set because the response has already started.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ThrowResponseAlreadyStartedException(String value)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.set_StatusCode(Int32 value)
   at PosBackend.Middlewares.ExceptionHandlingMiddleware.InvokeAsync(HttpContext context) in C:\Users\pinto\Documents\Development\POS\pos-backend\Middlewares\ExceptionHandlingMiddleware.cs:line 34
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
2025-02-06 22:45:44.550 +02:00 [WRN] The response has already started, the error page middleware will not be executed.
2025-02-06 22:45:44.558 +02:00 [ERR] Connection id "0HNA6TAPKH983", Request id "0HNA6TAPKH983:00000003": An unhandled exception was thrown by the application.
System.InvalidOperationException: StatusCode cannot be set because the response has already started.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ThrowResponseAlreadyStartedException(String value)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.set_StatusCode(Int32 value)
   at PosBackend.Middlewares.ExceptionHandlingMiddleware.InvokeAsync(HttpContext context) in C:\Users\pinto\Documents\Development\POS\pos-backend\Middlewares\ExceptionHandlingMiddleware.cs:line 34
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
   at Microsoft.WebTools.BrowserLink.Net.BrowserLinkMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Watch.BrowserRefresh.BrowserRefreshMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
2025-02-06 22:45:44.602 +02:00 [INF] Request finished HTTP/2 POST https://localhost:7005/api/auth/refresh - 401 null application/json 857.969ms
2025-02-06 22:46:14.591 +02:00 [INF] Request starting HTTP/2 GET https://localhost:7005/api/PricingPackages?pageNumber=1&pageSize=10 - null null
2025-02-06 22:46:14.612 +02:00 [INF] Failed to validate the token.
Microsoft.IdentityModel.Tokens.SecurityTokenInvalidIssuerException: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
   at Microsoft.IdentityModel.Tokens.Validators.ValidateIssuerAsync(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.Tokens.Validators.ValidateIssuer(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.Tokens.InternalValidators.ValidateAfterSignatureFailed(SecurityToken securityToken, Nullable`1 notBefore, Nullable`1 expires, IEnumerable`1 audiences, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateSignature(JsonWebToken jwtToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateSignatureAndIssuerSecurityKey(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateJWSAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
2025-02-06 22:46:14.616 +02:00 [ERR] ❌ Authentication failed: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
2025-02-06 22:46:14.617 +02:00 [INF] Bearer was not authenticated. Failure message: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
2025-02-06 22:46:14.637 +02:00 [INF] Authorization failed. These requirements were not met:
DenyAnonymousAuthorizationRequirement: Requires an authenticated user.
2025-02-06 22:46:14.643 +02:00 [ERR] An unhandled exception has occurred while executing the request.
System.InvalidOperationException: StatusCode cannot be set because the response has already started.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ThrowResponseAlreadyStartedException(String value)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.set_StatusCode(Int32 value)
   at PosBackend.Middlewares.ExceptionHandlingMiddleware.InvokeAsync(HttpContext context) in C:\Users\pinto\Documents\Development\POS\pos-backend\Middlewares\ExceptionHandlingMiddleware.cs:line 34
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
2025-02-06 22:46:14.645 +02:00 [WRN] The response has already started, the error page middleware will not be executed.
2025-02-06 22:46:14.646 +02:00 [ERR] Connection id "0HNA6TAPKH983", Request id "0HNA6TAPKH983:00000005": An unhandled exception was thrown by the application.
System.InvalidOperationException: StatusCode cannot be set because the response has already started.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ThrowResponseAlreadyStartedException(String value)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.set_StatusCode(Int32 value)
   at PosBackend.Middlewares.ExceptionHandlingMiddleware.InvokeAsync(HttpContext context) in C:\Users\pinto\Documents\Development\POS\pos-backend\Middlewares\ExceptionHandlingMiddleware.cs:line 34
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
   at Microsoft.WebTools.BrowserLink.Net.BrowserLinkMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Watch.BrowserRefresh.BrowserRefreshMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
2025-02-06 22:46:14.649 +02:00 [INF] Request finished HTTP/2 GET https://localhost:7005/api/PricingPackages?pageNumber=1&pageSize=10 - 401 null application/json 58.5512ms
2025-02-06 22:46:35.593 +02:00 [INF] Request starting HTTP/2 POST https://localhost:7005/api/auth/refresh - null 0
2025-02-06 22:46:35.595 +02:00 [INF] CORS policy execution failed.
2025-02-06 22:46:35.597 +02:00 [INF] Request origin https://localhost:7005 does not have permission to access the resource.
2025-02-06 22:46:35.610 +02:00 [INF] Failed to validate the token.
Microsoft.IdentityModel.Tokens.SecurityTokenInvalidIssuerException: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
   at Microsoft.IdentityModel.Tokens.Validators.ValidateIssuerAsync(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.Tokens.Validators.ValidateIssuer(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.Tokens.InternalValidators.ValidateAfterSignatureFailed(SecurityToken securityToken, Nullable`1 notBefore, Nullable`1 expires, IEnumerable`1 audiences, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateSignature(JsonWebToken jwtToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateSignatureAndIssuerSecurityKey(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateJWSAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
2025-02-06 22:46:35.612 +02:00 [ERR] ❌ Authentication failed: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
2025-02-06 22:46:35.613 +02:00 [INF] Bearer was not authenticated. Failure message: IDX10205: Issuer validation failed. Issuer: 'http://localhost:8383/realms/pisval-pos-realm'. Did not match: validationParameters.ValidIssuer: 'http://localhost:8282/realms/pisval-pos-realm' or validationParameters.ValidIssuers: 'null' or validationParameters.ConfigurationManager.CurrentConfiguration.Issuer: 'Null'. For more details, see https://aka.ms/IdentityModel/issuer-validation. 
2025-02-06 22:46:35.614 +02:00 [INF] Executing endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:46:35.615 +02:00 [INF] Route matched with {action = "RefreshToken", controller = "Auth"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.IActionResult] RefreshToken() on controller AuthController (PosBackend).
2025-02-06 22:46:35.617 +02:00 [INF] Executed action AuthController.RefreshToken (PosBackend) in 0.1688ms
2025-02-06 22:46:35.618 +02:00 [INF] Executed endpoint 'AuthController.RefreshToken (PosBackend)'
2025-02-06 22:46:35.619 +02:00 [ERR] An unhandled exception has occurred while executing the request.
System.InvalidOperationException: StatusCode cannot be set because the response has already started.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ThrowResponseAlreadyStartedException(String value)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.set_StatusCode(Int32 value)
   at PosBackend.Middlewares.ExceptionHandlingMiddleware.InvokeAsync(HttpContext context) in C:\Users\pinto\Documents\Development\POS\pos-backend\Middlewares\ExceptionHandlingMiddleware.cs:line 34
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
2025-02-06 22:46:35.620 +02:00 [WRN] The response has already started, the error page middleware will not be executed.
2025-02-06 22:46:35.621 +02:00 [ERR] Connection id "0HNA6TAPKH983", Request id "0HNA6TAPKH983:00000007": An unhandled exception was thrown by the application.
System.InvalidOperationException: StatusCode cannot be set because the response has already started.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ThrowResponseAlreadyStartedException(String value)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.set_StatusCode(Int32 value)
   at PosBackend.Middlewares.ExceptionHandlingMiddleware.InvokeAsync(HttpContext context) in C:\Users\pinto\Documents\Development\POS\pos-backend\Middlewares\ExceptionHandlingMiddleware.cs:line 34
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
   at Microsoft.WebTools.BrowserLink.Net.BrowserLinkMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Watch.BrowserRefresh.BrowserRefreshMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
2025-02-06 22:46:35.623 +02:00 [INF] Request finished HTTP/2 POST https://localhost:7005/api/auth/refresh - 401 null application/json 30.3932ms
